projects/batfish/src/main/java/org/batfish/bdp/BgpMultipathRib.java

```diff
@@ -47,6 +62,57 @@ public int comparePreference(BgpRoute lhs, BgpRoute rhs) {
     }
 
     /*
+     * AS path size is same. Now compare to best asPath (if available). Note we do not necessarily
+     * guarantee existing rhs route passes below test.
+     */
+    if (_bestAsPaths != null) {
+      AsPath bestAsPath = _bestAsPaths.get(lhs.getNetwork());
+      AsPath lhsAsPath = lhs.getAsPath();
+      AsPath rhsAsPath = rhs.getAsPath();
+      switch (_multipathEquivalentAsPathMatchMode) {
+        case EXACT_PATH:
+          if (bestAsPath.equals(lhsAsPath)) {
+            if (!bestAsPath.equals(rhsAsPath)) {
+              return 1;
+            }
+          } else if (bestAsPath.equals(rhsAsPath)) {
+            return -1;
+          }
+          break;
+
+        case FIRST_AS:
+          SortedSet<Integer> lhsFirstAsSet =
+              lhsAsPath.getAsSets().isEmpty()
+                  ? Collections.emptySortedSet()
+                  : lhsAsPath.getAsSets().get(0);
+          SortedSet<Integer> rhsFirstAsSet =
+              rhsAsPath.getAsSets().isEmpty()
+                  ? Collections.emptySortedSet()
+                  : rhsAsPath.getAsSets().get(0);
+          SortedSet<Integer> bestFirstAsSet =
+              bestAsPath.getAsSets().isEmpty()
+                  ? Collections.emptySortedSet()
+                  : bestAsPath.getAsSets().get(0);
+
+          if (bestFirstAsSet.equals(lhsFirstAsSet)) {
+            if (!bestFirstAsSet.equals(rhsFirstAsSet)) {
+              return 1;
+            }
+          } else if (bestFirstAsSet.equals(rhsFirstAsSet)) {
+            return -1;
+          }
+          break;
+
+        case PATH_LENGTH:
+          // Skip since already compared
+          break;
+        default:
+          throw new BatfishException(
+              "Unsupported " + MultipathEquivalentAsPathMatchMode.class.getName());
```

Comment:
Include the name of the class? `_multipathEquivalentAsPathMatchMode.getClass().getName()`?